### ISSUE TITLE
OpenTelemetry: Cap callback doesn't use parent Context

### ISSUE THREAD

**Issue by fusionat** [dotnetcore/CAP#1461]
Hello everyone,

### Overview of the Issue

We have implemented two services, each containing distinct methods.

Service A:

```
public IActionResult Start()
{
    _producer.Publish("sample.kafka.postgrsql", DateTime.Now.AddDays(14), "callbackme");
    return Ok("Service A is up and running.");
}

[CapSubscribe("callbackme")]
public IActionResult CallbackMe(string value)
{
    // Implementation details go here.
}

```
Service B:

```
[CapSubscribe("sample.kafka.postgrsql")]
public async Task<String> HandleEventAsync(DateTime value)
{
    return "Response received from the callback.";
}
```
### The Problem

We are encountering an issue where a new execution context, including a new TraceId, is generated in the CallbackMe method. We expected that the TraceId from the Start method would persist and be present in the CallbackMe method as well.

Strangely, this inconsistency with the TraceId only occurs when running our services in a Docker container. When testing locally outside of Docker, the TraceId remains consistent across all methods.

Could this be related to a multithreading issue?

**Comment by yang-xiaodong** [comment]
Hi,
It's a bug, fixed in commit 13a41d0d66674e9cf0f2d2e0306050dc43b5f02f

**Comment by yang-xiaodong** [comment]
Fixed in v8.0.1
